harness/determined/common/api/_session.py

```diff
@@ -1,27 +1,93 @@
-from typing import Any, Dict, Optional
+import abc
+import json as _json
+from typing import Any, Dict, Optional, Tuple, Union
 
 import requests
 import urllib3
 
+import determined as det
+from determined.common import requests as det_requests
 from determined.common import util
-from determined.common.api import authentication, certs, request
+from determined.common.api import authentication, certs, errors, request
 
 
-class Session:
-    def __init__(
-        self,
-        master: Optional[str],
-        user: Optional[str],
-        auth: Optional[authentication.Authentication],
-        cert: Optional[certs.Cert],
-        max_retries: Optional[urllib3.util.retry.Retry] = None,
-    ) -> None:
-        self._master = master or util.get_default_master_address()
-        self._user = user
-        self._auth = auth
-        self._cert = cert
-        self._max_retries = max_retries
+def _do_request(
```

Comment:
I'm not convinced that `determined.common.api.session` is a better home for this than `determined.common.api.request`, but I do really like your new `BaseSession._do_request`.

Reply:
My primary goal was to:
- remove the old one, as a way to assert that I did in fact update everywhere that called the old one
- make sure that the new location hints to developers that "this isn't the function you are looking for; you should be using a Session", especially developers who might go looking for where the old request function moved to thinking they could still use it.  And so for that, I really like it being inside `_session.py` and not importable from any accessible-looking namespace (like `det.common.api`)

Reply:
Thanks. This is good logic and a good explanation.

I'd personally count on `_do_request` looking private by the leading `_` in its name -- even in `det.common.api`. At this point it feels about 6 of one, but I'm leaving this open so we can touch on it when we chat.

Reply:
I do like it where it's at.  Where it is at is consistent with "all HTTP requests go through a Session", so it makes sense to me that the `_do_request()` function powering that statement is in `_session.py`.